### PR TITLE
Deprecate voting power

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

New fields representing the voting power of a neuron have recently been introduced to the governance canister (https://github.com/dfinity/ic/blob/release-2024-12-06_03-16-base/rs/nns/governance/canister/governance.did#L741-L743), and the original votingPower field has been marked as [deprecated](https://github.com/dfinity/ic/blob/d9fe2076f677a08734bed90c67b1c3f4056ed621/rs/nns/governance/canister/governance.did#L740). With this PR, `votingPower` is replaced by `decidedVotingPower`, which more accurately reflects the current voting power state of a neuron.

**Note**: The new fields are marked as optional to maintain backward compatibility of the type, but they are expected to always be provided.

# Changes

- Replace `NeuronInfo.votingPower` with `NeuronInfo.decidingVotingPower`.

# Tests

- Updated.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronVotingPowerSection.spec.ts [hunk 1]
```diff
@@ -126,7 +133,7 @@ describe("NnsStakeItemAction", () => {
     const po = renderComponent(neuron);
 
     expect(await po.getDescription()).toBe(
-      "voting_power = (30.00 + 0) × 1.00 × 1.06 × 0.50 = 6.14"
+      "voting_power = (30.00 + 0) × 1.00 × 1.06 × 0.50 = 3.07"
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/components/neuron-detail/NnsNeuronVotingPowerSection.spec.ts:136]: This change is an extension of the previous PR, where we updated the description but left the value untouched -https://github.com/dfinity/nns-dapp/pull/6010

File: frontend/src/lib/components/neuron-detail/NnsNeuronPageHeading.svelte [hunk 2]
```diff
@@ -52,7 +52,7 @@
       >
         {#if canVote}
           {replacePlaceholders($i18n.neuron_detail.voting_power_subtitle, {
-            $votingPower: formatVotingPower(neuron.votingPower),
+            $votingPower: formatVotingPower(neuron.decidingVotingPower ?? 0n),
```
Comment (yhabib) [inline frontend/src/lib/components/neuron-detail/NnsNeuronPageHeading.svelte:55]: out of curiosity, any idea why is this now optional?

Comment (yhabib) [reply frontend/src/lib/components/neuron-detail/NnsNeuronPageHeading.svelte:55]: [dummy question] what about `potential_voting_power`?  could we not use this value from the API instead of calculating it ourselves in `neuronPotentialVotingPower` ?

File: frontend/src/lib/utils/neuron.utils.ts [hunk 3]
```diff
@@ -109,8 +109,10 @@ const stateInfoMapper: StateMapper = {
 export const getStateInfo = (neuronState: NeuronState): StateInfo =>
   stateInfoMapper[neuronState];
 
+// TODO(mstr): Rename to neuronPotentialVotingPower
 /**
  * Calculation of the voting power of a neuron.
+ * ! Ignores the neurons activity state (votingPowerRefreshedTimestampSeconds)
```
Comment (yhabib) [inline frontend/src/lib/utils/neuron.utils.ts:115]: I don't get this comment, can you maybe rephrase it? 
does the initial exclamation mark have any meaning?

**Review by yhabib — COMMENTED** [review]
looks good, just some minor comments

File: frontend/src/lib/components/neuron-detail/NnsNeuronPageHeading.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/neuron-detail/NnsNeuronPageHeading.svelte:55]: They are optional because of the [candid definition of NeuronInfo](https://github.com/dfinity/ic-js/blob/d810d76c928b3d67839e168a4971e2f57622bd6a/packages/nns/candid/governance.did#L756-L757.) (by convention, all new fields are added as optional for backward compatibility, so the parser doesn’t break when trying to parse an older version), whereas the [votingPower](https://github.com/dfinity/ic-js/blob/d810d76c928b3d67839e168a4971e2f57622bd6a/packages/nns/candid/governance.did#L754) is not optional because it is an older value that was introduced before this convention.

Comment (mstrasinskis) [reply frontend/src/lib/components/neuron-detail/NnsNeuronPageHeading.svelte:55]: I think we calculate voting power only in cases where user input is provided. For example, when the user changes the neuron’s dissolve delay, we calculate it to display the potential change.

File: frontend/src/lib/utils/neuron.utils.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/lib/utils/neuron.utils.ts]: Good catch! The comment was half-baked. I’ve updated it.

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronVotingPowerSection.spec.ts (hunk 1 above)
Comment (yhabib) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronVotingPowerSection.spec.ts:136]: Not related but shouldn't the result be `15.9`

File: frontend/src/lib/utils/neuron.utils.ts [hunk 4]
```diff
@@ -163,6 +166,9 @@ interface VotingPowerParams {
   maxDissolveDelaySeconds?: number;
   minDissolveDelaySeconds?: number;
 }
+
+// TODO(mstr): Rename to calculatePotentialNeuronVotingPower
+// Because it doesn't take into account the the NNS neuron's activity state.
```
Comment (yhabib) [inline frontend/src/lib/utils/neuron.utils.ts:171]:
```suggestion
// Because it doesn't take into account the NNS neuron's activity state.
```

**Review by yhabib — APPROVED** [review]
some minor things

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronVotingPowerSection.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronVotingPowerSection.spec.ts:136]: Actually, this is not important in this test, because for the right part of the “equation,” we always display what is delivered by the API (decidingVotingPower).
